test/parallel/test-fs-watch-recursive-promise.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+// fs-watch on folders have limited capability in AIX.
+// The testcase makes use of folder watching, and causes
+// hang. This behavior is documented. Skip this for AIX.
+
+if (common.isAIX)
+  common.skip('folder watch capability is limited in AIX.');
+
+const { randomUUID } = require('crypto');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs/promises');
+const fsSync = require('fs');
+
+const tmpdir = require('../common/tmpdir');
+const testDir = tmpdir.path;
+tmpdir.refresh();
+
+(async function run() {
+  // Add a file to already watching folder
+
+  const testsubdir = await fs.mkdtemp(testDir + path.sep);
+  const file = `${randomUUID()}.txt`;
+  const filePath = path.join(testsubdir, file);
+  const watcher = fs.watch(testsubdir, { recursive: true });
+
+  let interval;
+
+  process.nextTick(common.mustCall(() => {
+    interval = setInterval(() => {
+      fsSync.writeFileSync(filePath, 'world');
+    }, 500);
+  }));
```

Comment:
We want to add the listener _before_ the await call, otherwise we won't be able to detect if the interval has been cleaned up... unless the interval has been cleaned up already.
```suggestion
  let interval;
  process.on('exit', function() {
    assert.ok(interval === null, 'watcher Object was not closed');
  });

  process.nextTick(common.mustCall(() => {
    interval = setInterval(common.mustCallAtLeast(() => {
      fsSync.writeFileSync(filePath, 'world');
    }), 50);
  }));
```